lib/arborist/build-ideal-tree.js

```diff
@@ -326,18 +337,11 @@ module.exports = cls => class IdealTreeBuilder extends cls {
     return Promise.all(add.map(s => {
       // in global mode, `npm i foo.tgz` needs to be resolved from
       // the current working dir, NOT /usr/local/lib!
-      const spec = npa(s, this[_global] ? process.cwd() : this.path)
-      // if it's just @'' then we reload whatever's there, or get latest
-      // if it's an explicit tag, we need to install that specific tag version
-      const isTag = spec.rawSpec && spec.type === 'tag'
-      return spec.name && !isTag ? spec : pacote.manifest(spec).then(mani => {
-        // if it's a tag type, then we need to run it down to an actual version
-        if (isTag)
-          return npa(`${mani.name}@${mani.version}`)
-
-        spec.name = mani.name
-        return spec
-      })
+      const currpath = this[_global] ? process.cwd() : this.path
```

Comment:
@isaacs I decided to leave this here for now, we can revisit it later

Reply:
We can land on a separate commit, but ought to get it fixed, may as well pull in with this changeset.

Reply:
eh, let's land it in a sep commit, maybe even better sep PR, let's thing about future us trying to bisect something 😂